core/chains/evm/config/chain_scoped_gas_estimator.go

```diff
@@ -138,8 +139,10 @@ func (d *daOracleConfig) OracleAddress() *types.EIP55Address {
 
 // CustomGasPriceCalldata returns the calldata for a custom gas price API.
 func (d *daOracleConfig) CustomGasPriceCalldata() string {
-	// TODO: CCIP-3710 update once custom calldata oracle is added
-	return ""
+	if d.c.OracleType == oracletype.CustomCalldata {
+		return d.c.CustomGasPriceCalldata
+	}
+	panic("CustomGasPriceCalldata is only supported on the CustomCalldata oracle type")
```

Comment:
I'm reluctant to panic here even if this should never happen. Maybe we can instead have a validation like `if daOracleConfig.CustomGasPriceCalldata() == "" { return err }` in the custom calldata oracle initialization to make sure this config can't be misused?